### PR TITLE
1.2.13 stepfixes

### PR DESCRIPTION
**Pull request by RomanIovlev**

### CODE REVIEW

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
- Added 11
           

Complexity decreasing per file
==============================
+ jdi-light/src/main/java/com/epam/jdi/light/elements/complex/table/DataTable.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247&bid=18347813)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java [hunk 1]
```diff
@@ -155,7 +154,7 @@ private boolean isDisplayed(T element) {
      */
     @JDIAction("Assert that '{name}' is not empty")
     public DataListAssert<T, D> notEmpty() {
-        jdiAssert(element.isEmpty() ? "list is empty" : "list is not empty", Matchers.is("list is not empty"));
+        jdiAssert(element().isEmpty() ? "list is empty" : "list is not empty", Matchers.is("list is not empty"));
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:157]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

File: jdi-light-html/src/main/java/com/epam/jdi/light/ui/html/actions/HtmlActions.java [hunk 2]
```diff
@@ -5,6 +5,8 @@
 import org.aspectj.lang.annotation.*;
 
 import static com.epam.jdi.light.actions.ActionHelper.*;
+import static com.epam.jdi.light.common.Exceptions.*;
```
Comment (codacy-review) [inline jdi-light-html/src/main/java/com/epam/jdi/light/ui/html/actions/HtmlActions.java:8]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.common.Exceptions'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

File: jdi-light-html/src/main/java/com/epam/jdi/light/ui/html/actions/HtmlActions.java [hunk 3]
```diff
@@ -5,6 +5,8 @@
...
+import static com.epam.jdi.light.settings.WebSettings.*;
```
Comment (codacy-review) [inline jdi-light-html/src/main/java/com/epam/jdi/light/ui/html/actions/HtmlActions.java:9]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.settings.WebSettings'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java [hunk 4]
```diff
@@ -120,7 +119,7 @@
      */
     @JDIAction("Assert that '{name}' has at least one displayed element")
     public DataListAssert<T, D> displayed() {
-        jdiAssert(element.isDisplayed() ? "displayed" : "hidden", Matchers.is("displayed"));
+        jdiAssert(element().isDisplayed() ? "displayed" : "hidden", Matchers.is("displayed"));
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:122]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

File: jdi-light/src/main/java/com/epam/jdi/light/elements/complex/table/BaseTable.java [hunk 5]
```diff
@@ -233,9 +230,26 @@ public WebList webColumn(int colNum) {
...
+    }
+    public int jsRowIndexByName(String rowName) {
+        int index = getRowHeaderIndex();
+        if (index == -1)
+            index = 1;
+        List<String> rowHeader;
+        try {
+            rowHeader = jsColumn(index);
+            if (rowHeader.size() == 0)
+                throw new RuntimeException();
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/elements/complex/table/BaseTable.java:243]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java [hunk 6]
```diff
@@ -130,7 +129,7 @@
      */
     @JDIAction("Assert that '{name}' is hidden")
     public DataListAssert<T, D> hidden() {
-        jdiAssert(map(element, this::isDisplayed), everyItem(Matchers.is(false)));
+        jdiAssert(map(element(), this::isDisplayed), everyItem(Matchers.is(false)));
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:132]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java [hunk 7]
```diff
@@ -145,7 +144,7 @@ private boolean isDisplayed(T element) {
      */
     @JDIAction("Assert that '{name}' is empty")
     public DataListAssert<T, D> empty() {
-        jdiAssert(element.isEmpty() ? "list is empty" : "list is not empty", Matchers.is("list is empty"));
+        jdiAssert(element().isEmpty() ? "list is empty" : "list is not empty", Matchers.is("list is empty"));
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:147]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java [hunk 8]
```diff
@@ -110,7 +109,7 @@
     @JDIAction("Assert that '{name}' elements [{0}] are displayed")
     public DataListAssert<T, D> displayed(String... names) {
         for (String name : names)
-            jdiAssert(element.get(name).isDisplayed() ? name + "displayed" : "hidden", Matchers.is(name + "displayed"));
+            jdiAssert(element().get(name).isDisplayed() ? name + "displayed" : "hidden", Matchers.is(name + "displayed"));
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:112]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/BaseAssert.java [hunk 9]
```diff
@@ -2,10 +2,11 @@
 
 import com.epam.jdi.light.asserts.core.SoftAssert;
 import com.epam.jdi.light.elements.base.JDIBase;
-import com.epam.jdi.light.elements.interfaces.base.IBaseElement;
-import com.epam.jdi.light.elements.interfaces.base.JDIElement;
+import com.epam.jdi.light.elements.interfaces.base.*;
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/BaseAssert.java:5]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.elements.interfaces.base'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/BaseAssert.java [hunk 10]
```diff
@@ -2,10 +2,11 @@
...
 import com.epam.jdi.tools.func.JFunc1;
 
+import static com.epam.jdi.tools.ReflectionUtils.*;
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/BaseAssert.java:8]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.tools.ReflectionUtils'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

File: jdi-light/src/main/java/com/epam/jdi/light/elements/complex/IList.java [hunk 11]
```diff
@@ -1,7 +1,6 @@
 package com.epam.jdi.light.elements.complex;
 
-import com.epam.jdi.light.elements.interfaces.base.HasValue;
-import com.epam.jdi.light.elements.interfaces.base.IBaseElement;
+import com.epam.jdi.light.elements.interfaces.base.*;
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/elements/complex/IList.java:3]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.elements.interfaces.base'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

**Comment by mf-code** [comment]
[5446] - w3SuX5PF

**Comment by mf-code** [comment]
[5446] - Vk0wU4JD

**Comment by mf-code** [comment]
[5446] - Allure report on Netlify: https://5ecfb3cd38bc5d6f0589423f--musing-swartz-6fc060.netlify.app

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
- Added 10
           

Complexity decreasing per file
==============================
+ jdi-light/src/main/java/com/epam/jdi/light/elements/complex/table/DataTable.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247&bid=18347813)

File: jdi-light-html/src/main/java/com/epam/jdi/light/ui/html/actions/HtmlActions.java (hunk 2 above)
Comment (codacy-review) [inline jdi-light-html/src/main/java/com/epam/jdi/light/ui/html/actions/HtmlActions.java:8]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.common.Exceptions'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/BaseAssert.java (hunk 9 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/BaseAssert.java:5]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.elements.interfaces.base'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java (hunk 1 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:157]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

File: jdi-light/src/main/java/com/epam/jdi/light/elements/complex/IList.java (hunk 11 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/elements/complex/IList.java:3]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.elements.interfaces.base'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java (hunk 4 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:122]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java (hunk 6 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:132]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

File: jdi-light-html/src/main/java/com/epam/jdi/light/ui/html/actions/HtmlActions.java (hunk 3 above)
Comment (codacy-review) [inline jdi-light-html/src/main/java/com/epam/jdi/light/ui/html/actions/HtmlActions.java:9]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.settings.WebSettings'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/BaseAssert.java (hunk 10 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/BaseAssert.java:8]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.tools.ReflectionUtils'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java (hunk 8 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:112]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java (hunk 7 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:147]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5600247)

**Comment by mf-code** [comment]
[5458] - n3a1ZxTw

**Comment by mf-code** [comment]
[5458] - WbpQiKF7

**Comment by mf-code** [comment]
[5458] - Allure report on Netlify: https://5ed0f7c2a15f10e71b8c04a6--musing-swartz-6fc060.netlify.app

**Comment by AkioFaris** [comment]
Disabled wrong Codacy issues: 
- Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'
- Avoid unused imports such as 'com.epam.jdi.light.elements.interfaces.base'

File: jdi-light/src/main/java/com/epam/jdi/light/elements/complex/table/BaseTable.java [hunk 12]
```diff
@@ -195,12 +190,13 @@ private void validateRowIndex(int rowNum) {
         }
     }
     public WebList webRow(int columnIndex, String rowName) {
-        return webColumn(columnIndex).get(getRowIndexByName(rowName)).finds(fromCellToRow);
+        //return webColumn(columnIndex).get(getRowIndexByName(rowName)).finds(fromCellToRow);
```
Comment (AkioFaris) [inline jdi-light/src/main/java/com/epam/jdi/light/elements/complex/table/BaseTable.java:193]: Do we need this comment?

**Pull request merged**